securedrop_client/crypto.py

```diff
@@ -145,17 +144,14 @@ def _gpg_cmd_base(self) -> list:
         cmd.extend(['--trust-model', 'always'])
         return cmd
 
-    def import_key(self, source_uuid: UUID, key_data: str, fingerprint: str) -> None:
-        session = self.session_maker()
-        local_source = session.query(Source).filter_by(uuid=source_uuid).one()
-
-        logger.debug("Importing key with fingerprint %s", fingerprint)
-        self._import(key_data)
-
-        local_source.fingerprint = fingerprint
-        local_source.public_key = key_data
-        session.add(local_source)
-        session.commit()
+    def import_key(self, source: Source) -> None:
+        """
+        Imports a Source's GPG key.
+        """
+        logger.debug("Importing key for source %s", source.uuid)
+        if not source.public_key:
+            raise CryptoError(f"Could not import key: source {source.uuid} has no key")
+        self._import(source.public_key)
 
     def _import(self, key_data: str) -> None:
         '''Wrapper for `gpg --import-keys`'''
```

Comment:
nit: i think this comment is no longer true

Reply:
Well, it never was, I guess, since the argument is just `--import`, but this is where the import is happening. Which did you mean? :slightly_smiling_face:

Reply:
i assumed that it was at least at the point in which the comment was written and then was never updated, but yeah we don't wrap `gpg --import-keys` so I was thinking a more accurate comment would be nice to have.

Reply:
Updated.